examples/beam-search/beam-search.cpp

```diff
@@ -25,6 +25,8 @@
 #include <signal.h>
 #endif
 
+namespace {
```

Comment:
I prefer keeping the `static` keyword instead of the anonymous namespace. The namespace is the same thing and it does reduce repetition, but it's an indirection that I don't think is better. With `static` I immediately know the scope when I see the function declaration